File: nxshell/qtbased/deepin-system-upgrade/src/deepin-system-upgrade-daemon/pkg/application/application.go
// SPDX-FileCopyrightText: 2022- 2023 UnionTech Software Technology Co., Ltd.
//
// SPDX-License-Identifier: LGPL-3.0-or-later

package application

import "github.com/linuxdeepin/go-lib/dbusutil"

func ExportAppManager(service *dbusutil.Service) error {
	m := newManager(service)
	return service.Export(dbusPath, m)
}

File: nxshell/module/lingmo-system-upgrade/src/lingmo-system-upgrade-tool/src/tool/application.cpp
// SPDX-FileCopyrightText: 2022 - 2023 UnionTech Software Technology Co., Ltd.
//
// SPDX-License-Identifier: LGPL-3.0-or-later

#include "application.h"

void Application::handleQuitAction() {
    Q_EMIT appQuit();
}

Application* Application::m_instance = nullptr;

Application* Application::getInstance(int &argc, char **argv) {
    return m_instance = (m_instance == nullptr ? new Application(argc, argv) : m_instance);
}
